pkg/executor/update.go

```diff
@@ -351,7 +351,10 @@ func (*UpdateExec) handleErr(colName model.CIStr, rowIdx int, err error) error {
 	if types.ErrOverflow.Equal(err) {
 		return types.ErrWarnDataOutOfRange.GenWithStackByArgs(colName.O, rowIdx+1)
 	}
-
+	if types.ErrTruncatedWrongVal.Equal(err) && col != nil && col.ColumnInfo != nil && col.ColumnInfo.GetType() == mysql.TypeTimestamp {
```

Comment:
So the condition here allowed to be igmored is "timestamp column && truncated value", right?
Is this condition check consistent with MySQL behaviour?

Reply:
Yes, this is compatible with MySQL, except for the error message.

**MySQL 8.0.29 **

```sql
mysql> create table t(t timestamp);
Query OK, 0 rows affected (0.09 sec)

mysql> insert into t values("2000-01-01");
Query OK, 1 row affected (0.00 sec)

mysql> update ignore t set t=cast("2099-01-01" as date);
Query OK, 1 row affected, 1 warning (0.00 sec)
Rows matched: 1  Changed: 1  Warnings: 1

mysql> show warnings;
+---------+------+--------------------------------------------+
| Level   | Code | Message                                    |
+---------+------+--------------------------------------------+
| Warning | 1264 | Out of range value for column 't' at row 1 |
+---------+------+--------------------------------------------+
1 row in set (0.00 sec)

mysql> select * from t;
+---------------------+
| t                   |
+---------------------+
| 0000-00-00 00:00:00 |
+---------------------+
1 row in set (0.00 sec)

mysql> select version();
+-----------+
| version() |
+-----------+
| 8.0.29    |
+-----------+
```

**This PR**

```sql
mysql> create table t(t timestamp);
Query OK, 0 rows affected (0.02 sec)

mysql> insert into t values("2000-01-01");
Query OK, 1 row affected (0.00 sec)

mysql> update ignore t set t=cast("2099-01-01" as date);
Query OK, 1 row affected, 1 warning (0.01 sec)
Rows matched: 1  Changed: 1  Warnings: 1

mysql> show warnings;
+---------+------+-----------------------------------------+
| Level   | Code | Message                                 |
+---------+------+-----------------------------------------+
| Warning | 1292 | Incorrect timestamp value: '2099-01-01' |
+---------+------+-----------------------------------------+
1 row in set (0.00 sec)

mysql> select * from t;
+---------------------+
| t                   |
+---------------------+
| 0000-00-00 00:00:00 |
+---------------------+
1 row in set (0.00 sec)

mysql> select tidb_version();
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| tidb_version()                                                                                                                                                                                                                                                              |
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Release Version: v8.2.0-alpha-675-g38a5139c3e
Edition: Community
Git Commit Hash: 38a5139c3eae82d3e2ea5f82f6ea89ceeabb505f
Git Branch: fix-50308
UTC Build Time: 2024-07-30 09:04:16
GoVersion: go1.21.5
Race Enabled: false
Check Table Before Drop: false
Store: unistore |
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```